modules/gapi/samples/face_detection.cpp

```diff
@@ -123,22 +123,22 @@ struct Face {
             facesNMS.push_back(faces[idx]);
             std::vector<int> tmpIndices = indices;
             indices.clear();
-            const double area1 = (faces[idx].bbox.x2 - faces[idx].bbox.x1 + 1) *
-                (faces[idx].bbox.y2 - faces[idx].bbox.y1 + 1);
+            const float area1 = static_cast<float>(faces[idx].bbox.x2 - faces[idx].bbox.x1 + 1) *
+                static_cast<float>(faces[idx].bbox.y2 - faces[idx].bbox.y1 + 1);
             for (size_t i = 1; i < tmpIndices.size(); ++i) {
                 int tmpIdx = tmpIndices[i];
-                const double interX1 = std::max(faces[idx].bbox.x1, faces[tmpIdx].bbox.x1);
-                const double interY1 = std::max(faces[idx].bbox.y1, faces[tmpIdx].bbox.y1);
-                const double interX2 = std::min(faces[idx].bbox.x2, faces[tmpIdx].bbox.x2);
-                const double interY2 = std::min(faces[idx].bbox.y2, faces[tmpIdx].bbox.y2);
-
-                const double bboxWidth = std::max(0.0, (interX2 - interX1 + 1));
-                const double bboxHeight = std::max(0.0, (interY2 - interY1 + 1));
-
-                const double interArea = bboxWidth * bboxHeight;
-                const double area2 = (faces[tmpIdx].bbox.x2 - faces[tmpIdx].bbox.x1 + 1) *
-                    (faces[tmpIdx].bbox.y2 - faces[tmpIdx].bbox.y1 + 1);
-                double overlap = 0.0;
+                const float interX1 = std::max(static_cast<float>(faces[idx].bbox.x1), static_cast<float>(faces[tmpIdx].bbox.x1));
+                const float interY1 = std::max(static_cast<float>(faces[idx].bbox.y1), static_cast<float>(faces[tmpIdx].bbox.y1));
+                const float interX2 = std::min(static_cast<float>(faces[idx].bbox.x2), static_cast<float>(faces[tmpIdx].bbox.x2));
+                const float interY2 = std::min(static_cast<float>(faces[idx].bbox.y2), static_cast<float>(faces[tmpIdx].bbox.y2));
```

Comment:
probably `static_cast<float>` can be applied only once on the result of `std::min` or `std::max`.

Reply:
Done